File: contents/bogo_sort/code/php/bogo_sort.php
<?php
declare(strict_types=1);

function is_sorted(array $array): bool
{
    for ($i = 0, $count = count($array); $i < $count - 1; $i++) {
        if ($array[$i] > $array[$i + 1]) {
            return false;
        }
    }

    return true;
}

function bogo_sort(array $array): array
{
    while (!is_sorted($array)) {
        shuffle($array);
    }

    return $array;
}


$unsorted = [10, 7, 3, 1, 4, 8, 5, 6, 9, 2];
$bogo_sorted = bogo_sort($unsorted);

printf('Unsorted: %s', implode(',', $unsorted));
echo PHP_EOL;
printf('Sorted: %s', implode(',', $bogo_sorted));
echo PHP_EOL;

File: contents/bogo_sort/code/swift/bogosort.swift
import Foundation

func isSorted(inputArray: [Int]) -> Bool {
    for i in 0..<inputArray.count-1 {
        if inputArray[i] > inputArray[i+1] {
            return false
        }
    }

    return true
}

func bogoSort(sortArray: inout [Int]) -> [Int] {
    while(!isSorted(inputArray: sortArray)) {
        sortArray.shuffle()
    }

    return sortArray
}

func main() {
    var testArray = [4,5,123,24,34,-5]
    print(bogoSort(sortArray: &testArray))
}

main()
